src/s3/error.rs

```diff
@@ -0,0 +1,30 @@
+use std::fmt;
+
+use openssl::error::ErrorStack;
+use reqwest::Error as ReqwestError;
+
+pub enum Error {
```

Comment:
What do you think about implementing `std::error::Error` on this type? I think it'll reduce surprise if someday we decide to do this in something that returns `Box<dyn std::error::Error>`. I could see an argument for YAGNI too though.